python/lsst/meas/modelfit/display/cModelDisplay.py

```diff
@@ -0,0 +1,264 @@
+    jointIm :
+        Image created from the joint fit of the dev and exp components of the
+        CModel for the supplied record at the same pixel locations as subImage
+    """
+
+    dev, exp, jointDev, jointExp, refFlux = reconstructCModel(exposure, record,
+                                                              cModelConfig)
+    # Get exposure cutout
+    footBBox = record.getFootprint().getBBox()
+    subImage = afwImage.ImageD(afwImage.ImageD(exposure.getImage(), True),
+                               footBBox)
+
+    # Build the psf
+    shapeletPsfKey = shapelet.MultiShapeletFunctionKey(
+        record.schema["modelfit_DoubleShapeletPsfApprox"])
+    psfApprox = record.get(shapeletPsfKey)
+
+    # Build the dev Image from the shapelet function
+    devIm = afwImage.ImageD(footBBox)
+    dev = dev.convolve(psfApprox)
+    dev.evaluate().addToImage(devIm)
+
+    # Build the exp image from the shapelet function
+    expIm = afwImage.ImageD(footBBox)
+    exp = exp.convolve(psfApprox)
+    exp.evaluate().addToImage(expIm)
+
+    # Build the joint image from the shapelet function
+    jointIm = afwImage.ImageD(footBBox)
+    jointDev = jointDev.convolve(psfApprox)
+    jointExp = jointExp.convolve(psfApprox)
+    jointDev.evaluate().addToImage(jointIm)
+    jointExp.evaluate().addToImage(jointIm)
+
+    # Scale each image by the reference flux
+    devImArr = devIm.getArray()
+    devImArr *= refFlux
+    expImArr = expIm.getArray()
+    expImArr *= refFlux
+    jointImArr = jointIm.getArray()
+    jointImArr *= refFlux
+    return subImage, devIm, expIm, jointIm
+
+
+def reconstructCModel(exposure, record, cModelConfig):
+    """ Reconstruct the CModel for the given record
+
+    Parameters
+    ----------
+    exposure : `lsst.afw.image.Exposure`
+        Exposure object that contains the source which was  modeled
+    record : `lsst.afw.table.SourceRecord`
+        Record object which contains the measurements made on the source
+    cModelConfig : `lsst.meas.modelfit.CModel(SingleFrame/Forced)Config`
+        Configuration object of the CModel plugin used in the measurement
+        process
+
+    Returns
+    -------
+    devShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the dev component of CModel
+    expShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model fo the exp component of CModel
+    devJointShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the dev component of CModel jointly
+        fit with the exp component
+    expJointShapelet : `lsst.shapelet.multiShapeletFunction`
+        Multi-component shapelet model of the exp component of Cmodel jointly
+        fit with the dec component
+    fitSysToMeasSys.flux : 'double'
+        Reference flux used to construct coordinate transformation between
+        the pixel flux system and the model flux system
```

Comment:
I think it'd be better to hide `fitSysToMeasSys` entirely within `reconstructCModel`, and apply the reference scaling to the shapelet coefficients before returning the `MultiShapeletFunction` objects.